### DISCUSSION TITLE
SQL parameterization not working with entity manager or query builder in mssql

### DISCUSSION THREAD

**Discussion by RussellPaine** [mikro-orm/mikro-orm#5960] [Q&A]
### Describe the bug

When profiling the generated SQL from the entity manager the where conditions are not passed in as parameters. We would expect anything passed in there to become a parameter. 

Logging the output of the entity manager it is putting the values provided as raw SQL and profiling this is the case.

Logging the out put using the `createQueryBuilder` function it claims to be using parameters but when actually profiling our DB this is not the case at all. Is there something we are missing?

### Reproduction

All we have done is create a connection to our DB then fork the em and try to find an entity.
```
const dbCredentials = {
    entities: [Entity, Entity2],
    dbName: '{dbName}',
    user: '{user}',
    password: '{password}',
    port: {port},
    host: "{host}",
    logger: console.log.bind(console),
    debug: true,
}

const orm = await MikroORM.init(dbCredentials);
```

```
const fork = orm.em.fork();

const test = await fork.find(Entity, {
        entity_id: 1
    },
    {
        populate: ["Entity2"]
    }
);
```

Resulting in:
```
exec sp_executesql @statement=N'{...query}'
```
Only `@statement` is defined. No other parameters.

### What driver are you using?

@mikro-orm/mssql

### MikroORM version

6.3.6

### Node.js version

22.5.1

### Operating system

Windows

### Validations

- [X] Read the [Contributing Guidelines](https://github.com/mikro-orm/mikro-orm/blob/master/CONTRIBUTING.md).
- [X] Read the [docs](https://mikro-orm.io/docs).
- [X] Check that there isn't [already an issue](https://github.com/mikro-orm/mikro-orm/issues) that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/mikro-orm/mikro-orm/discussions) or join our [Discord](https://discord.gg/w8bjxFHS7X).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by B4nan** [comment]
Please provide a complete reproduction. [We have tests](https://github.com/mikro-orm/mikro-orm/blob/master/tests/EntityManager.mssql.test.ts), this is well covered, its definitely not just “run any query and observe”.

**Comment by mackstann** [comment]
I'm experiencing the same thing with Postgres. When I look at my Postgres slow query logs, the queries run through Knex have query parameters and I never see the parameter values in the logged SQL. The queries run through Mikro-ORM have the parameter values interpolated into the SQL. This is problematic when the values contain sensitive info or are large.

I'm on 5.9.8. Maybe I can see if upgrading makes any difference.
